File: app/src/main/java/com/fernandosierra/door2door/presentation/custom/LoaderView.java
package com.fernandosierra.door2door.presentation.custom;

import android.animation.Animator;
import android.animation.AnimatorListenerAdapter;
import android.content.Context;
import android.support.annotation.Nullable;
import android.util.AttributeSet;

import com.airbnb.lottie.LottieAnimationView;

public class LoaderView extends LottieAnimationView {

    public LoaderView(Context context) {
        super(context);
    }

    public LoaderView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public LoaderView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public void cancelAnimation(@Nullable Runnable onAnimationEnd) {
        if (onAnimationEnd == null) {
            cancelAnimation();
        } else {
            final AnimatorListenerAdapter animatorListener = new AnimatorListenerAdapter() {
                @Override
                public void onAnimationEnd(Animator animation) {
                    removeAnimatorListener(this);
                    onAnimationEnd.run();
                }
            };
            loop(false);
            addAnimatorListener(animatorListener);
        }
    }
}

File: app/src/main/java/com/fernandosierra/door2door/presentation/custom/LoaderView.kt
package com.fernandosierra.door2door.presentation.custom

import android.animation.Animator
import android.animation.AnimatorListenerAdapter
import android.content.Context
import android.util.AttributeSet
import com.airbnb.lottie.LottieAnimationView

class LoaderView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0)
    : LottieAnimationView(context, attrs, defStyleAttr) {

    fun cancelAnimation(callback: (() -> Unit)?) {
        if (callback == null) {
            cancelAnimation()
        } else {
            val animatorListener = object : AnimatorListenerAdapter() {
                override fun onAnimationEnd(animation: Animator?) {
                    removeAnimatorListener(this)
                    callback.invoke()
                }
            }
            loop(false)
            addAnimatorListener(animatorListener)
        }
    }
}